File: MyAddressBook-starter/app/src/main/java/com/example/android/myaddressbook/ContactDatabase.java
package com.example.android.myaddressbook;

import android.content.Context;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;

import com.example.android.dao.ContactDao;
import com.example.android.model.Contact;

@Database(entities = Contact.class, version = 1)
public abstract class ContactDatabase extends RoomDatabase {

    private static ContactDatabase INSTANCE = null;

    public abstract ContactDao contactDao();

    public static ContactDatabase getInMemoryDatabase(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                    ContactDatabase.class, "Sample.db")
                            .build();
        }
        return INSTANCE;
    }

    public static void destroyInstance() {
        INSTANCE = null;
    }
}

File: MyAddressBook-starter/app/src/main/java/com/example/android/myaddressbook/ContactDatabase.kt
package com.example.android.myaddressbook

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Contact::class], version = 3)
abstract class ContactDatabase: RoomDatabase() {
    abstract fun contactDao(): ContactDao


    companion object {
        private var INSTANCE: ContactDatabase? = null
        const val DBNAME = "ContactDatabase"

        fun getInstance(context: Context):ContactDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                        context.applicationContext,
                        ContactDatabase::class.java,
                        DBNAME)
                        .fallbackToDestructiveMigration()
                        .build()
                INSTANCE = instance
                instance
            }
        }
    }
}